main.tf

```diff
@@ -432,6 +441,18 @@ resource "aws_emr_cluster" "default" {
     }
   }
 
+  dynamic "kerberos_attributes" {
+    for_each = var.kerberos_enabled ? [local.kerberos_attributes] : []
+
+    content {
+      ad_domain_join_password              = kerberos_attributes.value.ad_domain_join_password
```

Comment:
why do we need the `kerberos_attributes ` local? why don't just use the variables here?

Reply:
I think this way it's more clear.
In `for_each` we need to have either empty list or list with items which will make dynamic block to populate values.
Conditional expression can only be in one line (no newlines) so creating map with multiple kv pairs in `for_each` line would be hard to read.
If we have to have an item in list then it makes perfect sense for that to be `kerberos_attributes` with same structure as `terraform` accepts. As such structure is already created then it should be used in `content` to keep consistency (as opposed to using `vars` directly).

Any other option?

Reply:
I agree with that.

What I was refering to is to do it like this:

```
for_each = var.kerberos_enabled ? ["true"] : []

content {
      ad_domain_join_password              = var.kerberos_ad_domain_join_password
     .....
```

which is just another way of doing it